include/zenohcxx/api.hxx

```diff
@@ -36,10 +36,24 @@ using namespace zenohcxx;
 class Session;
 class Value;
 
-/// Text error message returned in ``std::variant<T, ErrorMessage>`` retrun types.
-/// The message is a sting represented as ``zenoh::Value`` object
+/// Text error message returned in ``std::variant<T, ErrorMessage>`` return types.
+/// The message is a sting represented as ``zenoh::Value`` object. See also ``zenoh::expect`` function
+/// which eases access to the return value in this variant type.
 typedef z::Value ErrorMessage;
 
+/// Uility function which either returns the value or throws an exception with the ``zenoh::ErrorMessage`` value
+/// @param v the ``std::variant<T, zenoh:::ErrorMessage>`` value
+/// @return the value of type ``T`` if ``std::variant<T, zenoh::ErrorMessage>`` value contains it
+/// @throws  ``zenoh::ErrorMessage`` if ``std::variant<T, zenoh::ErrorMessage>`` value contains it
+template <typename T>
+inline T expect(std::variant<T, z::Value>&& v) {
```

Comment:
What if we have a `std::variant<Value, Value>`?

Reply:
replaced to index() == 1